src/raftstore/store/peer_storage.rs

```diff
@@ -53,7 +56,8 @@ pub struct PeerStorage {
     pub region: metapb::Region,
     pub local_state: RaftLocalState,
 
-    pub snap_state: SnapState,
+    snap_state: Arc<RwLock<SnapState>>,
+    snap_sched: Scheduler<SnapTask>,
     snap_mgr: SnapManager,
```

Comment:
Why not move snap worker into SnapManager. So we can only use snap_mgr all places.